.github/workflows/translation.yml

```diff
@@ -0,0 +1,275 @@
+            --name $OUTPUT_CONTAINER_NAME \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --auth-mode login
+      #################################
+      # Upload translation, if exists #
+      #################################
+      - name: "Upload Translation"
+        if: matrix.glossary
+        run: |
+          set -eux
+
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          sas=`az storage container generate-sas -n $GLOSSARY_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions dlrw --expiry $end -o tsv`
+          az storage blob upload \
+            --overwrite true \
+            --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} \
+            --container-name $GLOSSARY_CONTAINER_NAME \
+            --name ${{ matrix.glossary }} \
+            --file translations/${{ matrix.language-id }}/${{ matrix.glossary }} \
+            --sas-token $sas
+          glossary_sas=`az storage container generate-sas -n $GLOSSARY_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lre --expiry $end -o tsv`
+          glossary_uri_with_sas=https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$GLOSSARY_CONTAINER_NAME/${{ matrix.glossary }}?$glossary_sas
+          glossary_json=",\"glossaries\": [{\"glossaryUrl\":\"$glossary_uri_with_sas\",\"format\":\"CSV\"}]"
+          echo "glossary_json=$glossary_json" >> $GITHUB_ENV
+      - name: "Upload Translation"
+        if: ${{ ! matrix.glossary }}
+        run: |
+          set -eux
+          echo "glossary_json= " >> $GITHUB_ENV
+      ############################
+      # Call translate REST API  #
+      ############################
+
+      - name: "Translate document to language"
+        run: |
+          set -ux
+          end=`date -u -d "30 minutes" '+%Y-%m-%dT%H:%MZ'`
+          input_sas=`az storage container generate-sas -n $INPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lr --expiry $end -o tsv`
+          output_sas=`az storage container generate-sas -n $OUTPUT_CONTAINER_NAME --auth-mode login --as-user --account-name ${{ secrets.STORAGE_ACCOUNT_NAME }} --https-only --permissions lw --expiry $end -o tsv`
+
+          operation_id=$(az rest --method post \
+            --url ${{ secrets.TRANSLATOR_API_URI }}translator/text/batch/v1.0/batches \
+            --headers Ocp-Apim-Subscription-Key=${{ secrets.TRANSLATOR_API_KEY }} \
+            --body "{\"inputs\": [{\"source\": { \"sourceUrl\": \"https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$INPUT_CONTAINER_NAME/?$input_sas\"}, \"targets\": [{\"targetUrl\": \"https://${{ secrets.STORAGE_ACCOUNT_NAME }}.blob.core.windows.net/$OUTPUT_CONTAINER_NAME?$output_sas\",\"language\": \"${{ matrix.language-id }}\" $glossary_json }]}]}" \
```

Comment:
maybe from a json checked into the repo with placeholders?

Reply:
done, good one